test/full_sync_test.cpp

```diff
@@ -0,0 +1,602 @@
+
+    EXPECT_CALL(*mockExtDataIfaces, fetchRbmcCredentials())
+        .WillRepeatedly([]() -> sdbusplus::async::task<> { co_return; });
+
+    nlohmann::json jsonData = {
+        {"Files",
+         {{{"Path", ManagerTest::tmpDataSyncDataDir.string() + "/srcFile1"},
+           {"DestinationPath",
+            ManagerTest::tmpDataSyncDataDir.string() + "/destFile1"},
+           {"Description", "Parse test file"},
+           {"SyncDirection", "Passive2Active"},
+           {"SyncType", "Immediate"}},
+          {{"Path", ManagerTest::tmpDataSyncDataDir.string() + "/srcFile2"},
+           {"DestinationPath",
+            ManagerTest::tmpDataSyncDataDir.string() + "/destFile2"},
+           {"Description", "Parse test file"},
+           {"SyncDirection", "Passive2Active"},
+           {"SyncType", "Immediate"}},
+          {{"Path", ManagerTest::tmpDataSyncDataDir.string() + "/srcFile3"},
+           {"DestinationPath",
+            ManagerTest::tmpDataSyncDataDir.string() + "/destFile3"},
+           {"Description", "Parse test file"},
+           {"SyncDirection", "Passive2Active"},
+           {"SyncType", "Immediate"}},
+          {{"Path", "/path/to/src/srcFile4"},
+           {"DestinationPath",
+            ManagerTest::tmpDataSyncDataDir.string() + "/destFile4"},
+           {"Description", "Parse test file"},
+           {"SyncDirection", "Passive2Active"},
+           {"SyncType", "Immediate"}}}}};
+
+    commonJsonData["Files"][2]["SyncDirection"] = "Passive2Active";
+
+    std::ofstream cfgFile(dataSyncCfgDir / "common_test_config.json");
+    ASSERT_TRUE(cfgFile.is_open()) << "Failed to open " << dataSyncCfgDir;
+    cfgFile << commonJsonData;
+    cfgFile.close();
+
+    std::string srcFile0{commonJsonData["Files"][0]["Path"]};
+    std::string destFile0{commonJsonData["Files"][0]["Path"]};
+
+    std::string srcFile1{jsonData["Files"][0]["Path"]};
+    std::string destFile1{jsonData["Files"][0]["DestinationPath"]};
+
+    std::string srcFile2{jsonData["Files"][1]["Path"]};
+    std::string destFile2{jsonData["Files"][1]["DestinationPath"]};
+
+    std::string srcFile3{jsonData["Files"][2]["Path"]};
+    std::string destFile3{jsonData["Files"][2]["DestinationPath"]};
+
+    std::string srcFile4{jsonData["Files"][3]["Path"]};
+    std::string destFile4{jsonData["Files"][3]["DestinationPath"]};
+
+    writeConfig(jsonData);
+    sdbusplus::async::context ctx;
+
+    std::string data0{"Data written on the file0\n"};
+    ManagerTest::writeData(srcFile0, data0);
+
+    std::string data1{"Data written on the file1\n"};
+    ManagerTest::writeData(srcFile1, data1);
+
+    std::string data2{"Data written on the file2\n"};
+    ManagerTest::writeData(srcFile2, data2);
+
+    std::string data3{"Data written on the file3\n"};
+    ManagerTest::writeData(srcFile3, data3);
+
+    std::string data4{"Data written on the file4\n"};
+    // ManagerTest::writeData(srcFile4, data4);
```

Comment:
Why is commented? or Add some comment like why are you doing this? I guess you are trying to avoid creating this file so that the fullsync will fail?